docs/content/drafts/DropdownMenu2.mdx

```diff
@@ -0,0 +1,308 @@
+storybook: '/react/storybook?path=/story/composite-components-dropdownmenu2'
+description: Use DropdownMenu to select a single option from a list of menu options.
+---
+
+import {Box, Text, Avatar} from '@primer/react'
+import {DropdownMenu, ActionList} from '@primer/react/drafts'
+import {Props} from '../../src/props'
+import State from '../../components/State'
+import {CalendarIcon, IterationsIcon, NumberIcon, SingleSelectIcon, TypographyIcon} from '@primer/octicons-react'
+
+<br />
+
+<State default={1}>
+  {([selectedIndex, setSelectedIndex]) => {
+    const fieldTypes = [
+      {icon: TypographyIcon, name: 'Text'},
+      {icon: NumberIcon, name: 'Number'},
+      {icon: CalendarIcon, name: 'Date'},
+      {icon: SingleSelectIcon, name: 'Single select'},
+      {icon: IterationsIcon, name: 'Iteration'}
+    ]
+    const selectedType = fieldTypes[selectedIndex]
+    return (
+      <Box sx={{border: '1px solid', borderColor: 'border.default', borderRadius: 2, padding: 6}}>
+        <Text sx={{fontSize: 1, mb: 1}}>Select field type:</Text>
+        <DropdownMenu>
+          <DropdownMenu.Button aria-label="Select field type" leadingIcon={selectedType.icon}>
+            {selectedType.name}
+          </DropdownMenu.Button>
+          <DropdownMenu.Overlay width="medium">
+            <ActionList>
+              {fieldTypes.map((type, index) => (
+                <ActionList.Item
+                  key={index}
+                  selected={index === selectedIndex}
+                  onSelect={() => setSelectedIndex(index)}
+                >
+                  <type.icon /> {type.name}
+                </ActionList.Item>
+              ))}
+            </ActionList>
+          </DropdownMenu.Overlay>
+        </DropdownMenu>
+      </Box>
+    )
+  }}
+</State>
+
+<br />
+
+```js
+import {DropdownMenu} from '@primer/react/drafts'
+```
+
+<br />
+
+## Examples
+
+### Minimal example
+
+`DropdownMenu` ships with `DropdownMenu.Button` which is an accessible trigger for the overlay. It's recommended to compose `ActionList` with `DropdownMenu.Overlay`
+
+&nbsp;
+
+```javascript live noinline
+// import {DropdownMenu, ActionList} from '@primer/react/drafts'
+const {DropdownMenu, ActionList} = drafts // ignore docs silliness; import like that ↑
+
+const fieldTypes = [
+  {icon: TypographyIcon, name: 'Text'},
+  {icon: NumberIcon, name: 'Number'},
+  {icon: CalendarIcon, name: 'Date'},
+  {icon: SingleSelectIcon, name: 'Single select'},
+  {icon: IterationsIcon, name: 'Iteration'}
+]
+
+const Example = () => {
+  const [selectedIndex, setSelectedIndex] = React.useState(1)
+  const selectedType = fieldTypes[selectedIndex]
+
+  return (
+    <DropdownMenu>
+      <DropdownMenu.Button aria-label="Select field type" leadingIcon={selectedType.icon}>
+        {selectedType.name}
+      </DropdownMenu.Button>
+      <DropdownMenu.Overlay width="medium">
+        <ActionList>
+          {fieldTypes.map((type, index) => (
```

Comment:
Nit for casing consistency.

```suggestion
{fieldTypes.map(({icon: Icon, name}, index) => (
```

so that you can do 

`<Icon /> {name}` instead of `<type.icon /> {type.name}`

Reply:
Done!